src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -1523,21 +1609,30 @@ tryRaisingOpToStableHLO(Operation *op, IRMapping &mapping, OpBuilder &builder,
           affine::AffineValueMap(AffineMap::get(op->getContext()), {});
 
       for (auto &innerOp : forOp.getBody()->without_terminator()) {
-        if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps).failed())
+        if (tryRaisingOpToStableHLO(&innerOp, mapping, builder, maps, pc)
+                .failed())
           return failure();
       }
 
       Value newIvInBody =
           builder.create<stablehlo::AddOp>(forOp.getLoc(), ivInBody, step);
 
       SmallVector<Value> loopCarried = {newIvInBody};
+      for (auto iterArg : forOp.getRegionIterArgs())
+        loopCarried.push_back(mapping.lookup(iterArg));
```

Comment:
we need to map the yielded values here (also we can check that `maps[mapping.lookup(iterARgs)].getOperands() == maps[mapping.lookup(yieldedIterARgs)].getOperands()` to make sure the hypothesis that every it is dependent on every iv in the loop nest is true):

```suggestion
      for (auto [iterArg, yieldedIterArgs] : llvm::zip(
        forOp.getRegionIterArgs(),
        forOp.getBody()->getTerminator()->getOperands()))
          loopCarried.push_back(mapping.lookup(yieldedIterArgs));

```